src/main/java/com/iexec/common/chain/DealParams.java

```diff
@@ -0,0 +1,19 @@
+package com.iexec.common.chain;
+
+import io.reactivex.annotations.Nullable;
+import lombok.*;
+
+import java.util.List;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+@Getter
+public class DealParams {
+
+    // Note to dev: the naming of the variables here is important since it will be stored on-chain
+    // that is why the names are not java compliant (use of underscore and no upper case)
+    private String iexec_args;
+    private List<String> iexec_input_files;
```

Comment:
You can use iexecArgs and iexecInputFiles, the mapping is done automatically by ObjectMapper.

Reply:
The mapping is not working if iexecArgs and iexecInputFiles are used.

Reply:
I think the @JsonProperty is needed.

Reply:
ok good with the proper JsonProperty